amaranth_soc/memory.py

```diff
@@ -579,3 +597,166 @@ def decode_address(self, address):
             return assignment.decode_address((address - addr_range.start) // addr_range.step)
         else:
             assert False # :nocov:
+
+
+class MemoryMapAnnotation(meta.Annotation):
+    schema = {
+        "$schema": "https://json-schema.org/draft/2020-12/schema",
+        "$id": "https://amaranth-lang.org/schema/amaranth-soc/0.1/memory/memory-map.json",
+        "type": "object",
+        "properties": {
+            "name": {
+                "type": "string",
+            },
+            "addr_width": {
+                "type": "integer",
+                "minimum": 0,
+            },
+            "data_width": {
+                "type": "integer",
+                "minimum": 0,
+            },
+            "alignment": {
+                "type": "integer",
+                "minimum": 0,
```

Comment:
You should add a comment here explaining what the value means. (It is a power of 2, right? Where the alignment in words is `1 << alignment`.)